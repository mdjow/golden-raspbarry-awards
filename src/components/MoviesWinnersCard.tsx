"use client";

import { useState } from "react";

import { Card } from "./Card";
import { Input } from "./Input";
import { Table } from "./Table";
import { useMovies } from "../hooks/useMovies";
import { Movie } from "../types/Movie";

export function WinnerMoviesCard() {
  const [year, setYear] = useState("");

  const { movies } = useMovies({ page: 0, isWinner: true, year, size: 6 });

  const columns = [
    {
      renderHeader: () => <span>ID</span>,
      renderCell: (row: Movie) => <span>{row.id}</span>,
    },
    {
      renderHeader: () => <span>Year</span>,
      renderCell: (row: Movie) => <span>{row.year}</span>,
    },
    {
      renderHeader: () => <span>Title</span>,
      renderCell: (row: Movie) => <span>{row.title}</span>,
    },
  ];

  return (
    <Card>
      <h3>List movies winners by year</h3>
      <Input
        type="number"
        placeholder="Search by year"
        defaultValue={year}
        onChange={(e) => setYear(e.target.value)}
      />
      <Table columns={columns} data={movies}></Table>
    </Card>
  );
}
