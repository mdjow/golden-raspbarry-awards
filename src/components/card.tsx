"use client";

import styled from "styled-components";

export const Card = styled.div`
  display: flex;
  flex-direction: column;

  background: var(--bg-primary);
  backdrop-filter: blur(10px);
  border-radius: 0px 0px 4px 4px;
  box-shadow: 0 1px 3px 0 var(--shapes-dark);

  padding: 20px;
  min-height: 250px;
  min-width: 250px;

  h3 {
    font-weight: 600;
    font-size: 18px;
    line-height: 200%;
  }

  p {
    font-weight: 400;
    font-size: 18px;
    line-height: 150%;
  }
`;
