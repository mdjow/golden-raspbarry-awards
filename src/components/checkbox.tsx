import styled from "styled-components";

export const Checkbox = styled.input.attrs({ type: "checkbox" })`
  height: 15px;
  margin: 6px 0;
`;
