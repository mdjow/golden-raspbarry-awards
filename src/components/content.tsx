"use client";

import styled from "styled-components";

export const Content = styled.div`
  background-color: var(--main-background-color);
  display: flex;
  flex: 1;
`;
