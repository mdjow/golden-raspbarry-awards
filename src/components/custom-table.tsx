import React from "react";
import styled from "styled-components";

interface Column<T> {
  renderHeader: () => React.ReactNode;
  renderCell: (row: T) => React.ReactNode;
}

interface CustomTableProps<T> {
  columns: Column<T>[];
  data?: T[];
}

const Table = styled.table`
  width: 100%;
  border-collapse: collapse;
  margin-bottom: 20px;

  th,
  td {
    border: 1px solid #ccc;
    padding: 10px;
    text-align: left;
  }

  th {
    background-color: #f2f2f2;
    text-align: center;
  }

  th * {
    width: 100%;
  }
`;

export function CustomTable<T>({ columns, data }: CustomTableProps<T>) {
  if (!data?.length) {
    return <center>no results found</center>;
  }

  return (
    <Table>
      <thead>
        <tr>
          {columns.map((column, index) => (
            <th key={index}>{column.renderHeader()}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {data.map((row, rowIndex) => (
          <tr key={rowIndex}>
            {columns.map((column, colIndex) => (
              <td key={colIndex}>{column.renderCell(row)}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </Table>
  );
}
