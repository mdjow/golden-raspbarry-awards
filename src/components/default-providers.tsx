"use client";

import { ReactNode } from "react";
import { QueryClientProvider } from "react-query";
import { ThemeProvider } from "styled-components";

import { queryClient } from "../services/queryClient";

interface DefaultProvidersProps {
  children: ReactNode;
}

const theme = {
  desktopBreakpoint: "968px",
  tableBreakpoint: "768px",
};

export function DefaultProviders({ children }: DefaultProvidersProps) {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </QueryClientProvider>
  );
}
