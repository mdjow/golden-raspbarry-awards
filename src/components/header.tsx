"use client";

import styled from "styled-components";
import { Saira_Stencil_One } from "next/font/google";

const sairaStencilOne = Saira_Stencil_One({
  weight: ["400"],
  subsets: ["latin"],
});

const MainHeader = styled.header`
  background-color: var(--header-background-color);
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 8px;
`;

const HeaderName = styled.a`
  color: var(--header-name-color);
  font-weight: 400;
  font-size: 26px;
  line-height: 150%;
`;

const Header: React.FC = () => {
  return (
    <MainHeader>
      <HeaderName className={sairaStencilOne.className}>
        Golden Raspberry Awards
      </HeaderName>
    </MainHeader>
  );
};

export default Header;
