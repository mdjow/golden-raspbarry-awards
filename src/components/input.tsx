import styled from "styled-components";

export const Input = styled.input<{ $inputColor?: string }>`
  padding: 6px;
  margin: 6px 0;
  color: ${(props) => props.$inputColor};
  border-color: var(--shapes-dark);
  border-radius: 3px;
`;
