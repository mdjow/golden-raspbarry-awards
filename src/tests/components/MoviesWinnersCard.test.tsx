import renderer from "react-test-renderer";

import { WinnerMoviesCard } from "../../components/MoviesWinnersCard";
import { winnerMoviesMock } from "./mocks/movieProjections.mock";

jest.mock("../../hooks/useMovies", () => ({
  useMovies: jest.fn(() => ({
    ...winnerMoviesMock,
  })),
}));

test("WinnerMoviesCard component matches snapshot", () => {
  const component = renderer.create(<WinnerMoviesCard />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
