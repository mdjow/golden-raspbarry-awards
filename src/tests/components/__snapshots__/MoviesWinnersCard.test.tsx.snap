// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnerMoviesCard component matches snapshot 1`] = `
<div
  className="sc-beyTiQ nILcX"
>
  <h3>
    List movies winners by year
  </h3>
  <input
    className="sc-guDMob cWLpcD"
    defaultValue=""
    onChange={[Function]}
    placeholder="Search by year"
    type="number"
  />
  <table
    className="sc-hLQTFJ iYFgiW"
  >
    <thead>
      <tr>
        <th>
          <span>
            <span>
              ID
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              Year
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              Title
            </span>
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span>
            1
          </span>
        </td>
        <td>
          <span>
            2020
          </span>
        </td>
        <td>
          <span>
            Movie A
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            2
          </span>
        </td>
        <td>
          <span>
            2021
          </span>
        </td>
        <td>
          <span>
            Movie B
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            3
          </span>
        </td>
        <td>
          <span>
            2022
          </span>
        </td>
        <td>
          <span>
            Movie C
          </span>
        </td>
      </tr>
      <tr />
    </tbody>
  </table>
</div>
`;
